<?xml version="1.0"?>
<!-- Xacro for GR1T2 -->
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <!--                   -->
    <!-- Imported elements -->
    <!--                   -->
    <xacro:include filename="$(find human_description)/robots/common/utils.xacro" />

    <xacro:macro name="robot_sensors"
        params="
        robot_id
        description_pkg:='human_description'
        ros2_control_plugin:='gazebo'
        connected_to:='base_footprint'
        xyz:='0 0 0'
        rpy:='0 0 0'
        inertials
        kinematics
        sensors_config
        with_sc:=false">

        <xacro:property name="gazebo"
            value="$(eval 'true' if ros2_control_plugin == 'gazebo' else 'false')" />

        <!-- IMU -->
        <!-- <xacro:dummy-link name="pelvis_imu_frame" />
        <xacro:robot-joint joint_name="pelvis_imu"
            joint_type="fixed"
            parent="${connected_to}"
            child="pelvis_imu_frame"
            joint_limits="${joint_limits}"
            kinematics="${kinematics}"
            inertials="${inertials}"
            dynamics="${dynamics}" /> -->

        <xacro:dummy-link name="head_camera_color_optical_frame" />
        <xacro:robot-joint joint_name="head_camera_color_optical_fixed_joint"
            joint_type="fixed"
            parent="waist2_p_link"
            child="head_camera_color_optical_frame"
            joint_limits="${joint_limits}"
            kinematics="${kinematics}"
            inertials="${inertials}" />

        <xacro:dummy-link name="right_wrist_camera_color_optical_frame" />
        <xacro:robot-joint joint_name="right_wrist_camera_color_optical_fixed_joint"
            joint_type="fixed"
            parent="r_arm7_wrist_p_link"
            child="right_wrist_camera_color_optical_frame"
            joint_limits="${joint_limits}"
            kinematics="${kinematics}"
            inertials="${inertials}" />

        <!-- <xacro:dummy-link name="middle_camera_link" />
        <xacro:robot-joint joint_name="middle_camera_link_fixed_joint"
            joint_type="fixed"
            parent="base_link"
            child="middle_camera_link"
            joint_limits="${joint_limits}"
            kinematics="${kinematics}"
            inertials="${inertials}"
            dynamics="${dynamics}" /> -->

    </xacro:macro>

</robot>